.eslintrc.js

```diff
@@ -25,12 +25,14 @@ module.exports = {
   },
   rules: {
     curly: 'error',
+    'no-case-declarations': 'off', // This recommended rule is actually harmful instead of helpful
+    '@typescript-eslint/no-shadow': 'error',
     'import/no-unresolved': 'error',
     'import/consistent-type-specifier-style': ['error', 'prefer-top-level'],
     'react/jsx-uses-vars': 'error',
     'react/jsx-uses-react': 'error',
-    'no-use-before-define': 'off',
-    '@typescript-eslint/no-use-before-define': 'off', // TODO consider enabling this (currently it reports styles defined at the bottom of the file)
+    'no-use-before-define': 'error',
```

Comment:
I really don't think we should enable this rule. While it could be handy when working with worklets, it would overall lower the quality of our code.

I know we aren't really strict about this, but in my opinion code files in this order:
```TSX
// important thing

// less important thing

// helper function

// stylesheet
```

Improves code quality since you read code from top to bottom, instead of the opposite. Instead we could probably define a separate rule just for worklets (if possible).
cc @tomekzaw @piaskowyk

Reply:
I haven't turned this rule on in our example app because of stylesheet, so this particular element can be omitted.

Reply:
What do you mean by 'this particular element'?

Reply:
Fixed, didn't want to use word "stylesheet" twice in one sentence 😆

Reply:
Yes, but stylesheet is just a part of the problem.

Reply:
I agree with Tomek. I don't see any benefits to using this rule either. If there are any, please let me know. However, I think we should disable this rule.